pkg/reconciler/build/build.go

```diff
@@ -352,6 +370,64 @@ func (c *Reconciler) conditionForPod(pod *corev1.Pod, stepsCompleted []string) c
 	}
 }
 
+func (c *Reconciler) finalize(ctx context.Context, build *buildapi.Build) error {
+	if !slices.Contains(build.GetFinalizers(), BuildFinalizer) {
+		return nil
+	}
+
+	if build.Finished() {
+		keychain, err := c.KeychainFactory.KeychainForSecretRef(ctx, registry.SecretRef{
+			ServiceAccount: build.Spec.ServiceAccountName,
+			Namespace:      build.Namespace,
+		})
+		if err != nil {
+			return err
+		}
+
+		if err := c.RegistryClient.Delete(keychain, build.Status.LatestImage); err != nil {
```

Comment:
We would also need to delete all of the additional tags as well.

Reply:
A build can push multiple images (see additionalTags). There is also the signed images and image attestations but that might be too much to try to delete and we probably don’t want to delete the attestations

Reply:
According to [docs](https://github.com/buildpacks-community/kpack/blob/main/docs/build.md#status), the `status.latestImage` field contain the image's digest. It should be enough to delete just this (no necessity to delete all tags individually): https://github.com/opencontainers/distribution-spec/blob/main/spec.md#deleting-manifests.

> Once deleted, a GET to /v2/<name>/manifests/<digest> and any tag pointing to that digest will return a 404.